privacyidea/lib/utils/compare.py

```diff
@@ -62,25 +87,92 @@ def _compare_contains(left, comparator, right):
         raise CompareError(u"Left value must be a list, not {!r}".format(type(left)))
 
 
+def _compare_matches(left, comparator, right):
+    """
+    Return True if ``left`` matches the regular expression given in ``right``.
+    Raise a CompareError if ``right`` is not a valid regular expression, or
+    if any other matching error occurs.
+    :param left: a string
+    :param right: a regular expression
+    :return: True or False
+    """
+    try:
+        return re.match("^" + right + "$", left) is not None
```

Comment:
Maybe we should make clear that the regexp must describe the complete string, not just a part (since we add `^` and `$`).

Reply:
Sounds good! I adjusted the docstring and description in c93c15aa1f1c2118cd5bfa60828364001537b7b

Reply:
That was a lie, it is actually 9c93c15aa1f1c2118cd5bfa60828364001537b7b